File: leetcode_75/Array_String/python/238.py
from typing import List


class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        pref1 = [1]
        pref2 = [1]
        for i in range(n):
            pref1.append(pref1[i] * nums[i])
            pref2.append(pref2[i] * nums[n - i - 1])
        pref2.reverse()

        result = list()
        for i in range(n):
            result.append(pref1[i] * pref2[i + 1])

        return result

File: leetcode_75/Array_String/js/238.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function (nums) {
    let n = nums.length
    let pref1 = [1]
    let pref2 = [1]
    for (let i = 0; i < n; ++i) {
        pref1.push(pref1[i] * nums[i])
        pref2.push(pref2[i] * nums[n - i - 1])
    }
    pref2.reverse()

    let result = []
    for (let i = 0; i < n; ++i) {
        result.push(pref1[i] * pref2[i + 1])
    }

    return result
};